### PR TITLE
Text to binary

### PR DESCRIPTION
**Pull request by EdmundKorley**
Here so we've gotten the first part! We are successfully outputting a 'bit matrix' of an input string. To test, you can run `make encode` to compile an `./encode` executable. Feed it an input string and it will output a 'bit matrix' where each row represents the base 2 equivalent of a specific character's ASCII value:

``` sh
$ make encode
# compiles
$ ./encode "abc"
01100001 # 97 or a
01100010 # 98 or b
01100011 # 99 or c
```

Its important that the rows in the matrix be the same width and that width be even. This is because in the next step we are to read two bits at a time and code that to a DNA base (maybe 00 => A, 01 => C, 10 => G, 11 => T).

### CODE REVIEW

**Comment by hlpostman** [comment]
Good comments, nice padding function.

File: srcs/text_2_bin_str.c [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	int		j;
+	char	*bit_str;
+
+	bin = 0;
+	i = 1;
+	j = 0;
+	bit_str = malloc(ASCII_BITS_MAXLEN + 2); // This needs to be an even # for the triplet conversion (sans \0).
+	while (ascii)
+	{
+		leftover = ascii % 2;
```
Comment (emkk) [inline srcs/text_2_bin_str.c]: This needs to be 4 instead of 2.

File: srcs/text_2_bin_str.c [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+		ascii /= 2;
```
Comment (EdmundKorley) [inline srcs/text_2_bin_str.c:49]: In extracting digit as a given base, we need to modulo by the base we want. In moving to parse ASCII as base-4, we need to change the 2s here (in lines 48 and 49) to 4.

File: includes/text2dna.h [hunk 3]
```diff
@@ -0,0 +1,15 @@
+// A set of useful headers.
+
+#ifndef TEXT2DNA_H
+#define TEXT2DNA_H
+
+#include <string.h>
+#include <stdio.h>
+#include <stdlib.h>
+#define ASCII_BITS_MAXLEN 7 // E.G. 127 is 111111 (7 digits)
```
Comment (EdmundKorley) [inline includes/text2dna.h:9]: We used this macro constant to decide how much to `malloc` and feed into our `ft_leftpad` function. This is considering the max ASCII value we will have to deal with is 127, which is 1333 in base 4. This is four digits so we should change this constant to 4. We should also rename this to be consistent, like `ASCII_BASE7_MAXLEN`.

**Pull request merged**